### PR TITLE
Enable creation of preview XML files from other collections.

### PR DESCRIPTION
**Pull request by Natim**
r? @leplatrem

### CODE REVIEW

File: kinto_amo/__init__.py [hunk 1]
```diff
@@ -15,14 +16,33 @@ def includeme(config):
...
-        settings.get('amo.addons', DEFAULT_ADDONS))
-    resources['plugins'] = parse_resource(
-        settings.get('amo.plugins', DEFAULT_PLUGINS))
-    resources['gfx'] = parse_resource(
-        settings.get('amo.gfx', DEFAULT_GFX))
-    resources['certificates'] = parse_resource(
-        settings.get('amo.certificates', DEFAULT_CERTIFICATES))
+
+    # Configure default settings
+    resources['blocklist'] = OrderedDict()
```
Comment (leplatrem) [inline kinto_amo/__init__.py]: Can we comment here why we use an ordereddict? (even if it says it's for reproducible tests)

File: kinto_amo/__init__.py [hunk 2]
```diff
@@ -15,14 +16,33 @@ def includeme(config):
...
+    for settings_key, settings_value in settings.items():
+        if not settings_key.startswith('amo.'):
+            continue
+        parts = settings_key.split('.')
+
+        if len(parts) == 2:  # amo.addons
+            prefix = 'blocklist'
+            _, blocklist = parts
+        else:  # amo.preview.addons
+            _, prefix, blocklist = parts
```
Comment (leplatrem) [inline kinto_amo/__init__.py]: ...or it will fail if `blocklist` contains a `.`

File: kinto_amo/__init__.py [hunk 3]
```diff
@@ -15,14 +16,33 @@ def includeme(config):
...
+
+        resource = resources.setdefault(prefix, OrderedDict())
+        resource[blocklist] = parse_resource(settings_value)
+
+    for resource in resources.values():
+        inter = set(resource.keys()) & {'addons', 'plugins', 'gfx', 'certificates'}
+        if len(inter) != 4:
+            raise ConfigurationError("Invalid blocklist: %s" % inter)
```
Comment (leplatrem) [inline kinto_amo/__init__.py]: Maybe it should say `Incomplete blocklist configuration` ?

File: kinto_amo/tests/test_views.py [hunk 4]
```diff
@@ -193,3 +196,56 @@ def test_not_enabled_gfx_are_ignored(self):
...
+
+    def get_app_settings(self, extras=None):
+        settings = super(AMOCustomTest, self).get_app_settings(extras)
+        settings['amo.preview.addons'] = '/buckets/blocklists/collections/addons'
+        settings['amo.preview.plugins'] = '/buckets/blocklists/collections/plugins'
+        settings['amo.preview.gfx'] = '/buckets/blocklists/collections/gfx'
+        settings['amo.preview.certificates'] = '/buckets/blocklists/collections/certificates'
+        return settings
+
+    # 1. Have a 404 when the resource is not defined
```
Comment (leplatrem) [inline kinto_amo/tests/test_views.py]: Maybe redundant with test title?

File: kinto_amo/__init__.py (hunk 2 above)
Comment (Natim) [inline kinto_amo/__init__.py]: Then it will fail again if blocklist contains a . on line 35.
I don't think we support . in the blocklist name.

Comment (Natim) [inline kinto_amo/__init__.py]: But at least we will be able to set it up.

**Pull request merged**